src/bmi_lgar.cxx

```diff
@@ -48,6 +56,20 @@ Initialize (std::string config_file)
 
 }
 
+/**
+ * @brief Allocate (or reallocate) storage for soil parameters
+ * 
+ */
+void BmiLGAR::realloc_soil(){
+  if(state->lgar_bmi_params.soil_depth_wetting_fronts != nullptr)
+    delete [] state->lgar_bmi_params.soil_depth_wetting_fronts;
+  if(state->lgar_bmi_params.soil_moisture_wetting_fronts != nullptr)
+    delete [] state->lgar_bmi_params.soil_moisture_wetting_fronts;
+
+  state->lgar_bmi_params.soil_depth_wetting_fronts = new double[state->lgar_bmi_params.num_wetting_fronts];
+  state->lgar_bmi_params.soil_moisture_wetting_fronts = new double[state->lgar_bmi_params.num_wetting_fronts];
+}
```

Comment:
As written, it seems like these could easily become `std::vector` and avoid the need for manual memory management.

Reply:
Agreed in principle, in practice, this would require modifying [lgar.cxx](https://github.com/NOAA-OWP/LGAR-C/blob/01dc874996c4fcaa748a43b49d5a8cb041698a96/src/lgar.cxx#L83) which is essentially extern `C` code masquerading as c++ code.

It is not clear if this project is intended to support the core lgar in C style code and provide a C++ BMI wrapper, or if it is intended to implement all of lgar in c++.  This needs to be a larger discussion outside the scope of fixing these memory leaks.